airbyte-webapp/src/components/connection/ConnectionOnboarding/ConnectionOnboarding.tsx

```diff
@@ -0,0 +1,256 @@
+            return (
+              <Tooltip
+                key={source?.sourceDefinitionId}
+                placement="right"
+                control={
+                  <button
+                    data-testid={`onboardingSource-${index}`}
+                    data-source-definition-id={source?.sourceDefinitionId}
+                    aria-label={tooltipText}
+                    className={styles.connectorButton}
+                    onClick={() => onCreate(source?.sourceDefinitionId)}
+                    onMouseEnter={() => setHighlightedSource(index as 0 | 1 | 2)}
+                  >
+                    <div className={styles.connectorIcon}>{getIcon(source?.icon)}</div>
+                  </button>
+                }
+              >
+                {tooltipText}
+              </Tooltip>
+            );
+          })}
+
+          <Tooltip
+            placement="right"
+            control={
+              <button
+                data-testid="onboardingSource-more"
+                className={styles.connectorButton}
+                onClick={() => onCreate()}
+                onMouseEnter={() => setHighlightedSource(3)}
+                aria-label={moreSourcesTooltip}
+              >
+                <PlusIcon className={styles.moreIcon} />
+              </button>
+            }
+          >
+            {moreSourcesTooltip}
+          </Tooltip>
+        </div>
+        <div className={styles.airbyte} aria-hidden="true">
+          <AirbyteIllustration sourceHighlighted={highlightedSource} destinationHighlighted={highlightedDestination} />
+        </div>
+        <div className={styles.destinations}>
+          <Text bold as="div" className={styles.destinationsTitle}>
+            <Tooltip
+              control={
+                <span>
+                  <FormattedMessage id="connection.onboarding.destinations" />{" "}
+                  <FontAwesomeIcon icon={faCircleQuestion} />
+                </span>
+              }
+            >
+              <FormattedMessage id="connection.onboarding.destinationsDescription" />
+            </Tooltip>
+          </Text>
+          {destinations.map((destination, index) => {
+            const tooltipText = formatMessage(
+              { id: "connection.onboarding.addDestination" },
+              { destination: destination?.name }
+            );
+            return (
+              <Tooltip
+                key={destination?.destinationDefinitionId}
+                placement="right"
+                control={
+                  <button
+                    className={styles.connectorButton}
+                    // onMouseEnter doesn't trigger on disabled buttons in React
+                    // https://github.com/facebook/react/issues/10109
+                    // Thus we just disable it via aria-disabled and make it non focusable via tabindex
+                    onMouseEnter={() => setHighlightedDestination(index as 0 | 1 | 2 | 3)}
```

Comment:
Seeing all the `as` makes me think this should just be a number.

Reply:
The type is probably the better choice, though I imagine `n <= 3 && n >= 0` isn't that bad performance-wise.
As long as we can get away with something like `type Index = 0 | 1 | 2 | 3`